subsys/net/lib/sockets/sockets_inet.c

```diff
@@ -160,10 +164,16 @@ int zsock_close_ctx(struct net_context *ctx)
 	ret = net_context_put(ctx);
 	if (ret < 0) {
 		errno = -ret;
-		return -1;
+		ret = -1;
 	}
 
-	return 0;
+	SYS_PORT_TRACING_OBJ_FUNC_EXIT(socket, close, sock, ret < 0 ? -errno : ret);
+
+	if (ret == 0) {
+		(void)sock_obj_core_dealloc(sock);
```

Comment:
Shouldn't this be done for all socket implementations, packet, TLS etc? `sock_obj_core_alloc()` is called for all socket types inside `z_impl_zsock_socket()`.

Reply:
Yep, I will fix that. TLS socket close calls `zsock_close()` directly so it does not seem to need any tweaking.

Reply:
My wooden eyes did not spot anything in upper layers that would need changing, could you give some more pointers what would need to change?